fix: Some fixes for #12850

MarryDream opened the pull request:
- refinement the error message when trueMail validation fails
- the settings of trueMail are not displayed after saving
- changing how `Active Email Validation` is saved

<!-- ℹ お読みください / README
PRありがとうございます！ PRを作成する前に、コントリビューションガイドをご確認ください:
Thank you for your PR! Before creating a PR, please check the contribution guide:
https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md
-->

## What
<!-- このPRで何をしたのか？ どう変わるのか？ -->
<!-- What did you do with this PR? How will it change things? -->
fix: Some fixes for #12850
- refinement the error message when trueMail validation fails
- the settings of trueMail are not displayed after saving
- changing how `Active Email Validation` is saved

## Why
<!-- なぜそうするのか？ どういう意図なのか？ 何が困っているのか？ -->
<!-- Why do you do it? What are your intentions? What is the problem? -->
Addressing some of the issues that arose with #12850

## Additional info (optional)
<!-- テスト観点など -->
<!-- Test perspective, etc -->

## Checklist
- [x] Read the [contribution guide](https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md)
- [x] Test working in a local environment
- [ ] (If needed) Add story of storybook
- [x] (If needed) Update CHANGELOG.md
- [ ] (If possible) Add tests

kakkokari-gtyih commented:
> refinement the error message when trueMail validation fails

Just run `pnpm build-misskey-js-with-types` 🙏

MarryDream commented:
> > refinement the error message when trueMail validation fails
> 
> Just run `pnpm build-misskey-js-with-types` 🙏

Got it, I've executed the command and re-committed it.

u1-liquid commented on packages/backend/src/core/EmailService.ts:
```diff
@@ -156,7 +156,7 @@ export class EmailService {
 	public async validateEmailForAccount(emailAddress: string): Promise<{
 		available: boolean;
-		reason: null | 'used' | 'format' | 'disposable' | 'mx' | 'smtp' | 'banned' | 'network' | 'blacklist';
+		reason: null | string;
```
I think this shouldn't be to changed 'just stirng', if there is another 'reason' value variants, please append it to here.

u1-liquid commented on packages/backend/src/core/EmailService.ts:
```diff
@@ -165,10 +165,17 @@ export class EmailService {
 			email: emailAddress,
 		});
 
+		if (exist !== 0) {
```
```suggestion
		if (exist > 0) {
```

u1-liquid commented on packages/backend/src/core/EmailService.ts:
```diff
@@ -185,33 +192,38 @@ export class EmailService {
 	private async verifyMail(emailAddress: string, verifymailAuthKey: string): Promise<{
 		valid: boolean;
-		reason: 'used' | 'format' | 'disposable' | 'mx' | 'smtp' | null;
+		reason: 'used' | 'format' | 'disposable' | 'mx' | 'smtp' | 'key' | null;
```
If the request failed due to the API key, wouldn't it be better to throw an error rather than showing it to the general user?

u1-liquid commented on packages/backend/src/core/EmailService.ts:
```diff
@@ -239,6 +252,12 @@ export class EmailService {
 			related_domains: string[];
 		};
 
+		if (json.message) {
```
Is 'every' message indicates 'api key' error?

MarryDream replied:
When the api usage exceeds the limit, will get `{ message: "error message content" }`, returned with only the `message` attribute. I've reworked the judgment method

u1-liquid commented on packages/backend/src/core/EmailService.ts:
```diff
@@ -284,22 +303,29 @@ export class EmailService {
-			
-			if (json.email === undefined || (json.email !== undefined && json.errors?.regex)) {
+
+			if (json.error) {
```
Is all 'error' indicates 'api key' error?

MarryDream replied:
When such an error occurs, the status code of the response will not be `200`. I looked at the implementation of the `httpRequestService` and it already throws an error in this case. So I removed the judgment here

u1-liquid commented on packages/misskey-js/src/autogen/apiClientJSDoc.ts:
```diff
@@ -1,6 +1,6 @@
 /*
  * version: 2023.12.2
- * generatedAt: 2024-01-02T08:53:57.449Z
+ * generatedAt: 2024-01-02T13:57:41.524Z
```
It seems that no actual changes have been made to the API, so it might be better not to have changes in this folder in this PR.

MarryDream replied:
Running `pnpm build-misskey-js-with-types` changed the timing of all type files. After updating the latest `develop` branch, I realized that someone else had already generated the correct types. So I'm now undoing the changes I made to the types files.

MarryDream committed:
fix: Some fixes for #12850

- refinement the error message when trueMail validation fails
- the settings of trueMail are not displayed after saving
- changing how `Active Email Validation` is saved

MarryDream commented:
@syuilo

syuilo commented:
🙏
